encoding/descaling.md

```diff
@@ -0,0 +1,452 @@
+or they will return odd graphs like this.
+
+For a change of pace,
+let's look at a good graph.
+
+![Aikatsu_Friend_NCOP_graph](images/descale_good_graph.png)
+```
+Output:
+Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33 
+Native resolution(s) (best guess): 810p
+```
+The results are very clear.
+There's a couple of deviations,
+but there's a very clear spike 
+going down to 810p.
+This is a good result
+for testing out varying kernels.
+
+## Descaling
+
+Now it's time to actually start descaling.
+Open up your Vapoursynth editor of choice,
+and import the clip:
+
+```Py
+src = lvf.src(r"BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
+```
+
+The next issue is figuring out
+what was used to [upsample](resampling.md#upsampling) the show.
+By default,
+getnative.py checks with Mitchell-Netravali
+(bicubic b=1/3, c=1/3).
+However, it might have also been upscaled
+using other kernels.
+
+Here is a list
+of some common kernels and values.
+
+* Lanczos
+* Spline
+* Bilinear
+* Bicubic b=1, c=0 (B-Spline)
+* Bicubic b=0, c=0 (Hermite)
+* Bicubic b=1/3, c=1/3 (Mitchell-Netravali)
+* Bicubic b=0, c=0.5 (Catmull-Rom)
+* Bicubic b=0, c=1 (Sharp Bicubic)
+
+The best way to figure out
+what is used is to simply try out
+a bunch of and use your eyes.
+Check for common scaling-related artifacting,
+like haloing,
+ringing,
+aliasing,
+etc.
+
+For bicubic it is important
+to keep in mind that
+you will typically find that
+the values match the following mathmatical expressions:
+
+* `b + 2c = 1`
+* `b + c = 1`
+* `b = 0, c = X`
+* `b = 1, c = 0`
+
+Whilst this isn't a 100% guarantee,
+this is the most common approach
+to resampling using bicubic,
+so it's worth keeping in mind.
+
+Here's an example of the previous frame
+when descaled using various kernels and settings
+(note that descale requires either GrayS, RGBS, or YUV444PS.
+I'll be using `split` and `join` from `kagefunc` to split the planes
+and then join them again in this example,
+and `get_w` from `vsutil` to calculate the width)[^1]:
+
+[Comparison between frames][manaria_compare]
+
+```Py
+import vapoursynth as vs
+import vsutil
+import kagefunc as kgf
+import fvsfunc as fvf
+core = vs.core
+
+src = core.lsmas.LWLibavSource("BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
+src = fvf.Depth(src, 32)
+
+Y, U, V = kgf.split(src)
+height = 878
+width = vsutil.get_w(height)
+
+descale_a = core.descale.Debilinear(Y, width, height).resize.Bilinear(1920, 1080)                                                        # Bilinear
+descale_b = core.descale.Debicubic(Y, width, height, b=1/3, c=1/3).resize.Bicubic(1920, 1080, filter_param_a=1/3, filter_param_b=1/3)    # Mitchell-Netravali
+descale_c = core.descale.Debicubic(Y, width, height, b=0, c=1).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1)            # Sharp Bicubic
+descale_d = core.descale.Debicubic(Y, width, height, b=1, c=0).resize.Bicubic(1920, 1080, filter_param_a=1, filter_param_b=0)            # B-Spline
```

Comment:
You’re just discarding all of these clips without ever saying why. The reader can’t know that you’re changing the `kgf.join()` call between comparisons.